framework/wazuh/agent.py

```diff
@@ -676,55 +679,107 @@ def get_outdated_agents(agent_list=None, offset=0, limit=common.database_limit,
+                   file_path=None, installer=None):
+    """Start the agent upgrade process.
 
+    Parameters
+    ----------
+    agent_list : list
+        List of agents ID's.
+    wpk_repo : str
+        URL for WPK download.
+    version : str
+        Version to upgrade to.
+    force : bool
+        force the update even if it is a downgrade.
+    use_http : bool
+        False for HTTPS protocol, True for HTTP protocol.
+    file_path : str
+        Path to the installation file.
+    installer : str
+        Selected installer.
 
-@expose_resources(actions=["agent:upgrade"], resources=["agent:id:{agent_list}"], post_proc_func=None)
-def get_upgrade_result(agent_list=None, timeout=3):
-    """Read upgrade result output from agent.
-
-    :param agent_list: List of agents ID's.
-    :param timeout: Maximum time for the call to be considered failed.
-    :return: Upgrade result.
+    Returns
+    -------
+    ID of created tasks
     """
-    # We access unique agent_id from list, this may change if and when we decide to add option to upgrade a list of
-    # agents
-    agent_id = agent_list[0]
+    result = AffectedItemsWazuhResult(all_msg='All upgrade tasks have been created',
+                                      some_msg='Some upgrade tasks have been created',
+                                      none_msg='No upgrade task has been created',
+                                      sort_fields=['task_id'], sort_ascending='True')
+
+    agent_list = list(map(int, agents_padding(result=result, agent_list=agent_list)))
+    if version and not version.startswith('v'):
+        version = f'v{version}'
+
+    agents_result_chunks = [agent_list[x:x + 100] for x in range(0, len(agent_list), 100)]
+
+    agent_results = list()
+    for agents_chunk in agents_result_chunks:
+        agent_results.append(
+            core_upgrade_agents(command='upgrade' if not (installer or file_path) else 'upgrade_custom',
+                                agents_chunk=agents_chunk, wpk_repo=wpk_repo, version=version, force=force,
+                                use_http=use_http, file_path=file_path, installer=installer))
+
+    for agent_result_chunk in agent_results:
+        for agent_result in agent_result_chunk['data']:
+            if agent_result['error'] == 0:
+                task_agent = {
+                    'agent': str(agent_result['agent']).zfill(3),
+                    'task_id': agent_result['task_id']
+                }
+                result.affected_items.append(task_agent)
+                result.total_affected_items += 1
+            else:
+                error = WazuhError(code=1810 + agent_result['error'], cmd_error=True,
+                                   extra_message=agent_result['message'])
+                result.add_failed_item(id_=str(agent_result['agent']).zfill(3), error=error)
+    result.affected_items = sorted(result.affected_items, key=lambda k: k['agent'])
```

Comment:
I think this is not the proper way to sort a distributed call

Reply:
Updated, now it is sorted when we receive the whole result from the api distributed call